docs/adrs/minimal-lifecycle.md

```diff
@@ -0,0 +1,164 @@
+off-chain.
+
+While the channel is opened, either participant may perform an **add** step.
+This does not change the stage of the channel. There is no limit to how many add
+steps can be performed. The participant locks more funds in the channel as
+further guarantee of their off-chain transacting.
+
+Once a participant no longer wishes to continue the arrangement they perform a
+**close** step. The channel is now **closed**. In the close the participant
+includes a summary of their off-chain transacting. They should no longer be
+transacting off-chain beyond this point. The time window for the other
+participant to provide their summary begins. There are two possible next steps.
+
+The other participant is able to perform a **resolve** step. They provide their
+summary to the L1, and can then unlock the funds they are due. The channel is
+now in the **resolved** stage.
+
+From the resolved stage, the participant that performed the close can finally
+**end** the channel. In doing, the channel is unstaged and all remaining funds
+are unlocked.
+
+Alternatively, the participant who performed the close step may perform an
+**elapse** step. This is valid only after the aforementioned time window has
+elapsed without the other participant performing a resolve. The step allows the
+participant to unlock the funds they demonstrated they were due. The channel is
+now in the **elapsed** stage.
+
+From the elapsed stage, the other participant can perform a **recover** step.
+This unstages the channel and unlocks the remaining funds.
+
+### Rationale
+
+## Discussion, Counter and Comments
+
+### Comments
+
+#### Less than a spec
+
+This document does not aspire to be detailed enough to be considered a spec. It
+is two establish the key terms and entities so that future discussions start
+with a shared vocabulary.
+
+#### Motivating terms
+
+Bitcoin Lightning inspires the use of the term 'close'. Beyond this, the framing
+is in terms of partially built transactions, rather the channel life-cycle.
+
+Other vocab was discussed. Choosing terms is a game of compromise that hopefully
+ends in between good enough and not bad. No terms borrowed from English will
+perfectly fit the precise shape of their employment in technical contexts like
+here.
+
+Some considered alternatives
+
+- live/dead for opened/closed. Sounded to zombie-esque.
+- phases for stages: the latter has stronger connotations of direction.
+
+#### Minimality
+
+A mutual close is not essential for a complete life-cycle. This will be left for
+a future ADR. The add step will also be left for a future ADR.
+
+It's been mooted that close includes the approved output address.
+
+#### Unilateral-ness
+
+The steps chosen are unilateral actions - they do not require any cooperation
+between the participants. Moreover the steps chosen expect a participant to look
+after only their own funds. The script is present to keep both participants
+safe.
+
```

Comment:
> signature associated with the final address 

The signature need not be associated to an address. You mean there is necessarily and associated address? I may want my funds paid there. I might want them somewhere else. That's up to me.

>  use "whitelisting" strategy

That sounds like a headache. Would this be an updateable or hardacoded? Are there not going to be circular dependencies if hardcoded? We really want to prevent double satisfaction, yes.

It seems unnecessarily complicated to try to reason about the "what ifs" that emerge from doing it in other ways we've considered. "Does a gateway use a different key for every channel? Can they use the same one? how do we prevent double satisfaction?" 

Please do a PR

Reply:
> The signature need not be associated to an address. You mean there is necessarily and associated address? I may want my funds paid there. I might want them somewhere else. That's up to me.

Let me clarify ask few clarifying questions:
* How do we authorize the `step`? If it is a signature under transaction then there is no difference between outputting UTxO at that address.
* If it is a signature under the redeemer itself then *if* user is capable of signing the tx itself using the key we again overcomplicating the flow from that perspective becuase it could express this address redirection *optionally* performing transaction which executes address replacement.

I believe that at the core we can support this multi stage closure but I don't see any reason to not allow simplified and more effective path as well.

> That sounds like a headache. Would this be an updateable or hardacoded? Are there not going to be circular dependencies if hardcoded? We really want to prevent double satisfaction, yes.

Double satisfaction is preventable using different mechanisms (indexing by order, indexing by datum) but none of seem to be 100% when mixed with arbitrary other scripts
because they can use or *not* use prevention at all.
I believe that we are trying to create an *efficient* baseline for a protocol which *should* possibly minimize obvious overheads on the L1. We can argue whether this execution paths are critical or rather corner cases but we have to be prepared that similar problem gonna repeat when we move to **mutual-close** which from my perspective should be really performant and allow batching.
BTW, doesn't mutual-close through multisig assume that the transaction is actually signed using the keys involved in the channel?p

Reply:
future adr